migrations/20210612_initial.sql

```diff
@@ -1,44 +1,57 @@
 CREATE TABLE contact_profiles ( -- remote user profile
   contact_profile_id INTEGER PRIMARY KEY,
-  contact_ref TEXT NOT NULL, -- this name must not contain spaces
-  properties TEXT NOT NULL DEFAULT '{}' -- JSON with user or contact profile
+  contact_ref TEXT NOT NULL, -- contact name set by remote user (not unique), this name must not contain spaces
+  properties TEXT NOT NULL DEFAULT '{}' -- JSON with contact profile properties
 );
 
--- the first record (id = 1) is reserved for the local user
+-- the first record (id = 1) is reserved for the first local user
 INSERT INTO contact_profiles (contact_profile_id, contact_ref) VALUES (1, '');
 
+
+CREATE TABLE users (
+  user_id INTEGER PRIMARY KEY,
+  contact_profile_id INTEGER NOT NULL UNIQUE REFERENCES contact_profiles -- user's profile
+);
+
+-- the first record (id = 1) is reserved for the first local user
+INSERT INTO users (user_id, contact_profile_id) VALUES (1, 1);
+
 CREATE TABLE known_servers(
   server_id INTEGER PRIMARY KEY,
   host TEXT NOT NULL,
   port TEXT NOT NULL,
   key_hash BLOB,
-  UNIQUE (host, port)
+  user_id INTEGER NOT NULL REFERENCES user_id,
+  UNIQUE (user_id, host, port)
 ) WITHOUT ROWID;
 
 CREATE TABLE contacts (
   contact_id INTEGER PRIMARY KEY,
-  contact_ref TEXT NOT NULL UNIQUE, -- contact name set by local user - must be unique
-  properties TEXT NOT NULL DEFAULT '{}', -- JSON set by local user
-  contact_profile_id INTEGER REFERENCES contact_profiles, -- sent by remote contact
-  contact_status TEXT NOT NULL DEFAULT ''
+  local_contact_ref TEXT NOT NULL UNIQUE, -- contact name set by local user - must be unique
+  local_properties TEXT NOT NULL DEFAULT '{}', -- JSON set by local user
+  contact_profile_id INTEGER UNIQUE REFERENCES contact_profiles, -- profile sent by remote contact, NULL for incognito contacts
```

Comment:
why not store nullable `contact_ref` and `properties` on `contacts` directly, and store user profiles (`properties`) in `users` table?